main/src/ca/uwaterloo/flix/language/phase/Eraser.scala

```diff
@@ -221,13 +246,34 @@ object Eraser {
       case Int16 => Int16
       case Int32 => Int32
       case Int64 => Int64
-      case Void | AnyType | Unit | BigDecimal | BigInt | String | Regex |
-           Region | Array(_) | Lazy(_) | Tuple(_) | MonoType.Enum(_) |
-           MonoType.Struct(_, _, _) | Arrow(_, _) | RecordEmpty | RecordExtend(_, _, _) | Native(_) | Null =>
+      case Void | AnyType | Unit | BigDecimal | BigInt | String | Regex | Region | Array(_) |
+           Lazy(_) | Tuple(_) | MonoType.Enum(_, _) | MonoType.Struct(_, _) | Arrow(_, _) |
+           RecordEmpty | RecordExtend(_, _, _) | Native(_) | Null =>
         MonoType.Object
     }
   }
 
+  private def erase(tpe: Type): Type = tpe match {
```

Comment:
Add comment

Reply:
I think both erase functions deserve some detailed comments to explain why there are two. And some examples to demonstrates what they translate.